zerver/middleware.py

```diff
@@ -210,12 +223,14 @@ class LogRequests(object):
     # for some views, process_view isn't run, so we call the start
     # method here too
     def process_request(self, request):
+        # type: (WSGIRequest) -> None
         request._log_data = dict()
         record_request_start_data(request._log_data)
         if connection.connection is not None:
             connection.connection.queries = []
 
     def process_view(self, request, view_func, args, kwargs):
+        # type: (WSGIRequest, Callable[..., None], List[str], Dict[str, Any]) -> None
```

Comment:
for `view_func`, I think it might be reasonable to just annotate it as `Callable,` -- this function doesn't access it at all.  But if we wanted to be precise, I think it's a `Callable[[HttpRequest, ...], HttpResponse]`